airgap/v2/cmd/client/download/download.go

```diff
@@ -0,0 +1,193 @@
+	DownloadCmd.MarkFlagRequired("output-directory")
+}
+
+func initLog() {
+	zapLog, err := zap.NewDevelopment()
+	if err != nil {
+		panic(fmt.Sprintf("Failed to initialize zapr, due to error: %v", err))
+	}
+	log = zapr.NewLogger(zapLog)
+}
+
+// initializeDownloadClient initializes the file retriever client based on provided configuration parameters
+func (dc *DownloadConfig) initializeDownloadClient() error {
+	// Fetch target address
+	address := viper.GetString("address")
+	if len(strings.TrimSpace(address)) == 0 {
+		return fmt.Errorf("target address is blank/empty")
+	}
+	log.Info(fmt.Sprintf("Target address: %v", address))
+
+	// Create connection
+	insecure := viper.GetBool("insecure")
+	var conn *grpc.ClientConn
+	var err error
+
+	if insecure {
+		conn, err = grpc.Dial(address, grpc.WithInsecure())
+	} else {
+		cert := viper.GetString("certificate-path")
+		creds, sslErr := credentials.NewClientTLSFromFile(cert, "")
+		if sslErr != nil {
+			return fmt.Errorf("ssl error: %v", sslErr)
+		}
+		opts := grpc.WithTransportCredentials(creds)
+		conn, err = grpc.Dial(address, opts)
+	}
+
+	// Handle any connection errors
+	if err != nil {
+		return fmt.Errorf("connection error: %v", err)
+	}
+
+	dc.client = fileretreiver.NewFileRetreiverClient(conn)
+	dc.conn = conn
+	return nil
+}
+
+// closeConnection closes the grpc client connection
+func (dc *DownloadConfig) closeConnection() {
+	if dc != nil && dc.conn != nil {
+		dc.conn.Close()
+	}
+}
+
+// downloadFile downloads the file received from the grpc server to a specified directory
+func (dc *DownloadConfig) downloadFile() error {
+	log.Info("Attempting to download file")
+
+	fn := strings.TrimSpace(dc.fileName)
+	fid := strings.TrimSpace(dc.fileId)
+	var req *fileretreiver.DownloadFileRequest
+	var name string
+
+	// Validate input and prepare request
+	if len(fn) == 0 && len(fid) == 0 {
+		return fmt.Errorf("file id/name is blank")
+	} else if len(fn) != 0 {
+		name = fn
+		req = &fileretreiver.DownloadFileRequest{
+			FileId: &v1.FileID{
+				Data: &v1.FileID_Name{
+					Name: fn},
+			},
+		}
+	} else {
+		name = fid
+		req = &fileretreiver.DownloadFileRequest{
+			FileId: &v1.FileID{
+				Data: &v1.FileID_Id{
+					Id: fid},
+			},
+		}
+	}
+
+	resultStream, err := dc.client.DownloadFile(context.Background(), req)
+	if err != nil {
+		return fmt.Errorf("failed to attempt download due to: %v", err)
+	}
+
+	var bs []byte
+	for {
+		file, err := resultStream.Recv()
+		if err == io.EOF {
+			log.Info("Server stream end")
+			break
+		} else if err != nil {
+			return fmt.Errorf("error while reading stream: %v", err)
+		}
+
+		data := file.GetChunkData()
+		if bs == nil {
+			bs = data
+		} else {
+			bs = append(bs, data...)
+		}
+	}
+
+	outFile, err := os.Create(dc.outputDirectory + string(os.PathSeparator) + name)
+	if err != nil {
+		return fmt.Errorf("error while creating output file: %v", err)
+	}
+
```

Comment:
```suggestion
defer outFile.Close()
```

We should defer just in case we error on write. Since this is a client it probably won't matter but it's a good practice.